hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/S3ATestUtils.java

```diff
@@ -818,6 +828,22 @@ public static void callQuietly(final Logger log,
     return null;
   }
 
+  /**
+   * Get a file status from S3A with the {@code needEmptyDirectoryFlag}
+   * state probed.
+   * This accesses a package-private method in the
+   * S3A filesystem.
+   * @param fs filesystem
+   * @param dir directory
+   * @return a status
+   * @throws IOException
+   */
+  public static S3AFileStatus getStatusWithEmptyDirFlag(
```

Comment:
just to remind me that I have a similar method in pr so I have to rebase: https://github.com/apache/hadoop/pull/1134/files#diff-d946bda31a9bf644846f57b45a89524dR831